lib/v2/screens/sign_up/viewmodels/signup_bloc.dart

```diff
@@ -113,12 +117,23 @@ class SignupBloc extends Bloc<SignupEvent, SignupState> {
 
     yield currentState.copyWith(addPhoneNumberState: AddPhoneNumberState.loading(currentAddPhoneNumberState));
 
-    final inviteSecret = state.claimInviteState.inviteModel!.inviteSecret;
+    final String inviteSecret = secretFromMnemonic(state.claimInviteState.inviteMnemonic!);
     final displayName = state.displayNameState.displayName;
     final username = state.createUsernameState.username;
 
+    EOSPrivateKey privateKey = EOSPrivateKey.fromRandom();
+
     final Result result = await _addPhoneNumberUseCase.run(
-        inviteSecret: inviteSecret!, displayName: displayName!, username: username!, phoneNumber: phoneNumber);
+      inviteSecret: inviteSecret,
+      displayName: displayName!,
+      username: username!,
+      privateKey: privateKey,
+      phoneNumber: phoneNumber,
+    );
+
```

Comment:
As soon as the account was created, we need to save the private key to settings so it doesn't get lost.